mesonbuild/dependencies/base.py

```diff
@@ -1847,13 +1848,21 @@ def _windows_sanitize_path(path: str) -> str:
         # Ensure that we use USERPROFILE even when inside MSYS, MSYS2, Cygwin, etc.
         if 'USERPROFILE' not in os.environ:
             return path
-        # Ignore executables in the WindowsApps directory which are
-        # zero-sized wrappers that magically open the Windows Store to
-        # install the application.
+        # The WindowsApps directory is a bit of a problem. It contains
+        # some zero-sized .exe files which have "reparse points", that
+        # might either launch an installed application, or might open
+        # a page in the Windows Store to download the application.
+        #
+        # To handle the case where the python interpreter we're
+        # running on came from the Windows Store, if we see the
+        # WindowsApps path in the search path, replace it with
+        # dirname(sys.executable).
         appstore_dir = Path(os.environ['USERPROFILE']) / 'AppData' / 'Local' / 'Microsoft' / 'WindowsApps'
         paths = []
         for each in path.split(os.pathsep):
-            if Path(each) != appstore_dir:
+            if Path(each) == appstore_dir and 'WindowsApps' in sys.executable:
+                paths.append(os.path.dirname(sys.executable))
+            else:
```

Comment:
So, 
```
if Path(each) == appstore_dir and 'WindowsApps' not in sys.executable:
    paths.append(each)
```

Seems broken. Either make it an if inside of an if, or make the else into an elif.

Reply:
oops, good point. fixed